blocks/embed/embed.js

```diff
@@ -0,0 +1,222 @@
+      <iframe src="https://www.youtube.com${vid ? `/embed/${vid}?rel=0&amp;v=${vid}` : embed}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" allow="encrypted-media; accelerometer; gyroscope; picture-in-picture" allowfullscreen="" scrolling="no" title="Content from Youtube" loading="lazy"></iframe>
+    </div>`;
+    return embedHTML;
+}
+  
+const embedInstagram = (url) => {
+    const location = window.location.href;
+    const src = `${url.origin}${url.pathname}${url.pathname.charAt(url.pathname.length - 1) === '/' ? '' : '/'}embed/?cr=1&amp;v=13&amp;wp=1316&amp;rd=${location.endsWith('.html') ? location : `${location}.html`}`;
+    const embedHTML = `<div style="width: 100%; position: relative; display: flex; justify-content: center">
+      <iframe class="instagram-media instagram-media-rendered" id="instagram-embed-0" src="${src}"
+        allowtransparency="true" allowfullscreen="true" frameborder="0" height="530" style="background: white; border-radius: 3px; border: 1px solid rgb(219, 219, 219);
+        box-shadow: none; display: block;" loading="lazy">
+      </iframe>
+    </div>`;
+    return embedHTML;
+}
+  
+const embedVimeo = (url) => {
+    const linkArr = url.href.split('/');
+    const video = linkArr ? linkArr[3] : linkArr;
+
+    const embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+        <iframe src=\"https:\/\/player.vimeo.com\/video\/${video}?app_id=122963\" 
+        style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" 
+        frameborder=\"0\" allow=\"autoplay; fullscreen; picture-in-picture\" allowfullscreen  
+        title="Content from Vimeo" loading="lazy"><\/iframe>
+    </div>`;
+    return embedHTML;
+}
+  
+const embedSpark = (url) => {
+    let embedURL = url;
+    if (!url.pathname.endsWith('/embed.html') && !url.pathname.endsWith('/embed')) {
+      embedURL = new URL(`${url.href}${url.pathname.endsWith('/') ? '' : '/'}embed.html`);
+    }
+  
+    return getDefaultEmbed(embedURL);
+}
+  
+const embedTwitter = (url) => {
+    const embedHTML = `<blockquote class="twitter-tweet"><a href="${url}"></a></blockquote>`;
+    loadScript('https://platform.twitter.com/widgets.js');
+    return embedHTML;
+}
+
+const embedTiktok = (url) => {
+    let resultHtml = document.createElement('div');
+    resultHtml.setAttribute('id', 'tiktok');
+    fetch(`https://www.tiktok.com/oembed?url=${url}`)
+    .then(response => response.json())
+    .then(data => {
+        loadScript('https://www.tiktok.com/embed.js');
+        const $tiktok = document.getElementById('tiktok')
+        $tiktok.outerHTML = data.html;
+    })
+    .catch((error) => {
+        console.error('Error:', error);
+    });
+
+    return resultHtml.outerHTML;
+}
+
+const embedSlidShare = (url) => {
+    let resultHtml = document.createElement('div');
+    resultHtml.setAttribute('id', 'slideShare');
+    fetch(url)
```

Comment:
that's awesome, i had no idea that this worked ;)
i thought we were stuck with a more complicated URL extraction from an authoring standpoint...

generally we have been using `await fetch()` so maybe we should use that notation for consistency sake